src/helpers/sendNotifications.js

```diff
@@ -0,0 +1,219 @@
+import Pusher from 'pusher';
+import dotenv from 'dotenv';
+import models from '../db/models';
+import EmailNotificationAPI from './EmailNotificationAPI';
+
+dotenv.config();
+
+const {
+  Art, Comment, Following, User
+} = models;
+
+const pusher = new Pusher({
+  appId: process.env.PUSHER_APPID,
+  key: process.env.PUSHER_KEY,
+  secret: process.env.PUSHER_SECRET,
+  cluster: process.env.PUSHER_CLUSTER,
+  encrypted: true
+});
+
+/** SendNotifications Controller Class */
+class SendNotifications {
+  /**
+   * @constructor
+   * @param {Object} payloadObject - Details of the notification.
+   */
+  constructor(payloadObject) {
+    this.payloadObject = payloadObject;
+  }
+
+  /**
+   * @Represents a newArticleNotification
+   * @param {Object} articleDetailsObject - The details of the article.
+   * @return {Object|string} success string OR Error message .
+   */
+  static async newArticleNotification(articleDetailsObject) {
+    const {
+      artId, artistId, slugifiedTitle, artTitle, artFeaturedImg, artDescription
+    } = articleDetailsObject;
+
+    const artistUsername = await User.findOne({
+      where: { id: artistId },
+      attributes: ['username'],
+    });
+
+    const followers = await Following.findAll({
+      where: { userId: artistId },
+      include: [
+        {
+          model: User,
+          attributes: ['id', 'email'],
+        }
+      ],
+      attributes: [],
+    });
+
+    const mailList = [];
+
+    followers.forEach((follower) => {
+      const { id, email } = follower.User;
+      mailList.push(email);
+
+      pusher.trigger('new-article', `article-${artId}-${id}-event`, {
+        message:
+          `<a href="${process.env.APP_BASE_URL}/api/v1/arts/${slugifiedTitle}">
+        New Article: ${artTitle} <br> by ${artistUsername}</a>`
+      });
+    });
+
+    const sendEmailNotifications = new EmailNotificationAPI({
+      to: '"Team Merry 👻" <noreplyteammerry@gmail.com>',
+      bcc: mailList,
+      subject: `New Article: ${artTitle} by ${artistUsername}`,
+      message: `<h3>New Article: ${artTitle} <br> 
+by ${artistUsername}</h3>
+<h6>Check it out here:</h6>
+<p><img src="${artFeaturedImg}" width="450px" height="200px"
+alt="Featured Image"></p>
+<p><a href="http://${process.env.APP_BASE_URL}/api/v1/arts/${slugifiedTitle}"> 
+${artTitle} </a><br> by ${artistUsername}</p>
+<p>${artDescription}</p>`
+    });
+
+    return sendEmailNotifications.sendEmail();
+  }
+
+  /**
+   * @Represents getUserInfo
+   * @param {number} userID - The id of the user.
+   * @return {Object} get User's Username
+   */
+  static async getUserInfo(userID) {
```

Comment:
This looks like a middleware, what do you think?

Reply:
Yes I agree, but this file, sendNotifications.js is a helper class. How may I extract `getUserInfo()` as a middleware? because its being used only in this class.